js/chips.js

```diff
@@ -19,67 +25,112 @@ function selectChip(chip) {
 }
 
 function clearSelection() {
-	const selectedChip = document.querySelectorAll('.selected'); // check the DOM for elements with a `selected` class
-	if (selectedChip) { // if there are any instances of the `selected` class...
-		for (var i = 0, len = selectedChip.length; i < len; i++) { // loop through them...
-			selectedChip[i].classList.remove('selected'); // and remove the `selected` class
-		}
-	} else {
-		return;
-	}
+		selectedChip.classList.remove('selected'); // remove the `selected` class
+		selectedChip = undefined; // set the global variable back to undefined
+		return selectedChip; // return the undefined variable back to the global scope
 }
 
-function validateMoves(selectedChip, currentSquare) {
+function validateMoves(currentSquare) {
 	// things to check to find out what the available moves are for the selected chip:
 	// - get the array of nextValidSquares from the square's markup (or encapsulate that data into another object somewhere?)
-	// - check each of the nextValidQuares for empty squares or enemy chips/friendly chips using the squareContains function
+	// - check each of the nextValidQuares for empty squares or enemy chips/friendly chips using the evalSquare function
 	// - create an array of available empty valid squares that the chip can move to
 	// - create an array of valid squares that contain enemy chips
 	// - when a valid square contains an enemy chip, check for available capture
 	// - if a capture is available, create an array of valid squares to *jump-to* that would *capture* an enemy chip
 }
 
-function squareContains(selectedChip) {
-	// decide if a valid square is empty.
-	// if not empty, decide if the square contains a chip
-	// if the square contains a chip, find out if it is an enemy chip!
-	// isEnemyChip(selectedChip, evalChip)
-	// return possible values: "emptySquare", "friendlyChip", "enemyChip"
+function evalSquare(selectedSquare) {
+
+	// if the clicked square is the parent of the selectedChip, do nothing.
+	if (selectedSquare.children[0] == selectedChip) {
+		return false; 
+	}
+
+	var squareHas;
+
+	if (selectedSquare.children.length === 0) {
+		squareHas = "emptySquare";
+	} else if (isEnemyChip(selectedSquare.children[0])) {
+		squareHas = "enemyChip";
+	} else {
+		squareHas = "friendlyChip";
+	}
+	
+	return squareHas;
+
 }
 
-function isEnemyChip(selectedChip, evalChip) {
-	// check the selectedChip's color and test it against another chip, `evalChip` to decide if it is an enemy chip.
-	// return a boolean
+function isEnemyChip(evalChip) {
+	if (!selectedChip) {
+		return false;
+	}
+
+	var chipColor;
+	var evalChipColor;
+	
+	if (selectedChip.classList.contains("red")) {
+		chipColor = "red";
+	} else {
+		chipColor = "black";
+	}
+
+	if (evalChip.classList.contains("red")) {
+		evalChipColor = "red";
+	} else {
+		evalChipColor = "black";
+	}
+	
+	if (chipColor === evalChipColor) {
+		return false;
+	} else {
+		return true;
+	}
```

Comment:
this is a lot of lines of code to determine who is the enemy! Once we start using state management in this project (via Vue instances!) we won't need such a lengthy evaluation. We should (in the future) already know the current player's chip color. For now we'll use this function to progress to other functions.